lib/src/widget/asset_picker.dart

```diff
@@ -110,10 +114,24 @@ class AssetPickerState<Asset, Path> extends State<AssetPicker<Asset, Path>>
   @override
   void didChangeAppLifecycleState(AppLifecycleState state) {
     super.didChangeAppLifecycleState(state);
+
     if (state == AppLifecycleState.resumed) {
-      PhotoManager.requestPermissionExtend().then((PermissionState ps) {
-        widget.builder.permission.value = ps;
-      });
+      if (_checkPermissionCompleter == null) {
+        final completer = Completer<void>();
+        _checkPermissionCompleter = completer;
+        PhotoManager.requestPermissionExtend().then((PermissionState ps) {
+          widget.builder.permission.value = ps;
+          completer.complete();
+        }).catchError((Object e, StackTrace s) {
+          completer.completeError(e, s);
+        });
+      }
+    }
+
+    if (state == AppLifecycleState.paused) {
```

Comment:
Is the behavior correct? Why should we clear the pending status when paused?

Reply:
I have updated the logic to reset status, please check.

Reply:
Hmm. This is becoming more weird. What problem does the delay solves?